### PR TITLE
add backend, create thunks, api requests, fix some bugs

### PR DESCRIPTION
**Pull request by RuslanKuyanets**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -50,13 +55,12 @@ const App: React.FC<AppStatePropsType & AppDispatchPropsType> = (props) => {
     </div>
   );
 }
+//<PersistGate loading={null} persistor={persistor}></PersistGate>
```
Comment (IslamRustamov1) [inline src/App.tsx]: delete comment

File: src/App.tsx [hunk 2]
```diff
@@ -67,33 +71,37 @@ const mapStateToProps = (state: AppStateType): AppStatePropsType => {
...
+  removeTask: removeTask, 
+  toggleProgress: toggleProgress,
+  changeTask: changeTask,
+  toggleProgressAll: toggleProgressAll,
+  removeTaskAllComplited: removeTaskAllComplited,
+  changeFilter: todoActions.changeFilter,
+  getTodos: getTodosThunkCreator
 })(App)
-
+//
```
Comment (IslamRustamov1) [inline src/App.tsx]: delete comment

File: src/Redux/store.ts [hunk 3]
```diff
@@ -1,22 +1,23 @@
...
+import thunk from 'redux-thunk';
 
-const persistConfig = {
-    key: 'root',
-    storage: localStorage
-}
+// const persistConfig = {
+//     key: 'root',
+//     storage: localStorage
+// }
```
Comment (IslamRustamov1) [inline src/Redux/store.ts]: delete comment

File: src/Redux/store.ts [hunk 4]
```diff
@@ -1,22 +1,23 @@
...
     todo: todoReducer
 })
 
-const persistedReducer = persistReducer(persistConfig, reducers)
+// const persistedReducer = persistReducer(persistConfig, reducers)
 
-let store = createStore(persistedReducer)
-export let persistor = persistStore(store)
+let store = createStore(reducers, applyMiddleware(thunk))
+// export let persistor = persistStore(store)
```
Comment (IslamRustamov1) [inline src/Redux/store.ts]: delete comments

File: src/Redux/store.ts [hunk 5]
```diff
@@ -1,22 +1,23 @@
...
 const reducers = combineReducers({
     todo: todoReducer
 })
 
-const persistedReducer = persistReducer(persistConfig, reducers)
+// const persistedReducer = persistReducer(persistConfig, reducers)
 
-let store = createStore(persistedReducer)
-export let persistor = persistStore(store)
+let store = createStore(reducers, applyMiddleware(thunk))
```
Comment (IslamRustamov1) [inline src/Redux/store.ts:11]: use const

File: src/Redux/todo-reducer.ts [hunk 6]
```diff
@@ -113,5 +127,68 @@ export const todoActions = {
...
+            payload: todos
+        } as const
+    }
+}
+
+export const getTodosThunkCreator = (): ThunkType<TodoActionsType> => {
+    return async (dispatch) => {
+        dispatch(todoActions.toggleIsFetching(true))
+        const response = await todoApi.getTodos()
+        dispatch(todoActions.getTodos(response.reverse()))
```
Comment (IslamRustamov1) [inline src/Redux/todo-reducer.ts:149]: Response should be reversed

Comment (IslamRustamov1) [reply src/Redux/todo-reducer.ts:149]: on BE

**Review by IslamRustamov1 — APPROVED** [review]

**Comment by RuslanKuyanets** [comment]
good

**Pull request merged**